sqlglot/parser.py

```diff
@@ -7132,6 +7151,11 @@ def _parse_tokens(
 
         return this
 
+    def _parse_pipe_syntax_query(self, query: exp.Query) -> exp.Query:
```

Comment:
If I'm not mistaken we can only get a `Select` passed here, right? Via `_parse_from`:
```suggestion
    def _parse_pipe_syntax_query(self, query: exp.Select) -> exp.Query:
```

Reply:
Note: the return type being `Query` is fine. When we add support for `UNION`, etc, we could end up with a `Query` instance that is not a `Select`.